tests/functional/serving/test_experiments_resource.py

```diff
@@ -196,6 +196,154 @@ def test_unknown_parameter(self, client, ephemeral_storage):
+        }
+
+    def test_default_is_latest_version(self, client, ephemeral_storage):
+        """Tests that the latest experiment is returned when no version parameter exists"""
+
+        storage = ephemeral_storage.storage
+
+        _add_experiment(storage, name="a", version=1, _id=1)
+        _add_experiment(storage, name="a", version=3, _id=2)
+        _add_experiment(storage, name="a", version=2, _id=3)
+        _add_trial(storage, experiment=1, id_override="ae8", status="completed")
+        _add_trial(storage, experiment=2, id_override="ae8", status="reserved")
+        _add_trial(storage, experiment=3, id_override="ae8", status="interrupted")
+
+        response = client.simulate_get("/experiments/status/a")
+        response1 = client.simulate_get("/experiments/status/a?version=1")
+        response3 = client.simulate_get("/experiments/status/a?version=3")
+
+        assert response.status == "200 OK"
+        assert response1.status == "200 OK"
+        assert response3.status == "200 OK"
+
+        assert response.json == {
+            "trials_completed": 0,
+            "best_trials_id": None,
+            "best_evaluation": None,
+            "start_time": "0001-01-01 00:00:00",
+            "finish_time": "0001-01-01 00:00:00",
+            "max_trials": 10,
+            "nb_trials": 1,
+            "progress": 0.0,
+            "trial_status_count": {"reserved": 1},
+            "duration": "23:59:50",
+            "whole_clock_time": "23:59:50",
+            "eta": "infinite",
+            "eta_milliseconds": None,
+        }
+
+        assert response.json == response3.json
+        assert response.json != response1.json
+
+    def test_specific_version(self, client, ephemeral_storage):
+        """Tests that the specified version of an experiment is returned"""
+
+        storage = ephemeral_storage.storage
+
+        _add_experiment(storage, name="a", version=1, _id=1)
+        _add_experiment(storage, name="a", version=3, _id=2)
+        _add_experiment(storage, name="a", version=2, _id=3)
+        _add_trial(storage, experiment=1, id_override="ae8", status="completed")
+        _add_trial(storage, experiment=2, id_override="ae8", status="reserved")
+        _add_trial(storage, experiment=3, id_override="ae8", status="interrupted")
+
+        response = client.simulate_get("/experiments/status/a?version=2")
+        assert response.status == "200 OK"
+        assert response.json == {
+            "trials_completed": 0,
+            "best_trials_id": None,
+            "best_evaluation": None,
+            "start_time": "0001-01-01 00:00:00",
+            "finish_time": "0001-01-01 00:00:00",
+            "max_trials": 10,
+            "nb_trials": 1,
+            "progress": 0.0,
+            "trial_status_count": {"interrupted": 1},
+            "duration": "23:59:50",
+            "whole_clock_time": "23:59:50",
+            "eta": "infinite",
+            "eta_milliseconds": None,
+        }
```

Comment:
It would be good to add a tests with max_trials infinite so that if we change the value returned to something that json cannot handle we will detect it in the test.

Reply:
Ok !